core/src/Streamly/Internal/FileSystem/Path/Common.hs

```diff
@@ -463,3 +484,42 @@ append :: (Unbox a, Integral a) =>
     OS -> (Array a -> String) -> Array a -> Array a -> Array a
 append os toStr a b =
     withAppendCheck os toStr b (doAppend os a b)
+
+-- | Split a path into components separated by the path separator. "."
+-- components in the path are ignored except in the leading position. Multiple
+-- consecutive separators are ignored.
+--
+-- >>> :{
+--  splitPath Common.Posix = Stream.toList . fmap unpackPosix . Common.splitPath Common.Posix . packPosix
+--  splitPath Common.Windows = Stream.toList . fmap unpackWindows . Common.splitPath Common.Windows . packWindows
+-- :}
+--
+-- >>> splitPath Common.Posix "."
+-- ["."]
```

Comment:
Do the doctests pass.
How do we keep the leading `.`?
We are filtering them out right?

Perhaps `off /= 0` should be a part of `shouldFilterOut`
```
    shouldFilterOut (off, len) =
        len == 0 ||
            (len == 1 && unsafeIndexChar off arr == '.' && off /= 0)
```